### DISCUSSION TITLE
How not to display IP address on timeout error

### DISCUSSION THREAD

**Discussion by yu-fuku** [stepci/stepci#119] [Q&A]
Hi!

If the test fails with a timeout, the destination IP address is displayed in the error message.

```
FAIL  example

Summary

  ✕ GET request failed after 75.057s

⚠︎ example › GET request

connect ETIMEDOUT {IP address}
```

IP address is secret information for us and we don't want to display it on the console.
Is there any way to avoid displaying the IP address?

**Comment by mishushakov** [comment]
Interesting use-case 🤔
Have you tested it with the `--hide` option? Does it still display the IP?

I think the error comes from `got`, our HTTP client and we cannot change it easily
Maybe we should allow to hide certain types of errors, such as HTTP errors from the display?

Meanwhile, I'd suggest maybe proxying it through a tunnel, so you only see the tunnel address

**Comment by yu-fuku** [comment]
I tried `--hide` options, but it shows the IP address.

I don't think it is very effective because the IP address of proxy server is displayed even if it goes through proxy. 
(I don't think the risk of IP address leakage remains the same).

I understood that you are displaying `got` error message.
So I found it difficult to change the error message.

I think if you don't have use-case where you want to display IP address, I'd be happy to handle it with `got` error type (etc. ) and not display the IP address.
However, I am not that troubled about this issue, so there is no need to force them to add the feature.

Thank you for replay

**Reply by mishushakov** [reply]
You could just fork the runner and add a condition here:

https://github.com/stepci/runner/blob/c734591b3e354fd6f76367b8d2e35e2784af63d6/src/index.ts#L962-L967
